Migrate to click (#188)
jupyterlab/jupyter-ai: docs/source/users/index.md

- `openai-chat`
- `sagemaker-endpoint`

:::{warning}
As of v0.8.0, only the `%%ai` *cell* magic may be used to invoke a language
model, while the `%ai` *line* magic is reserved for invoking subcommands.
:::

### Listing available models

Jupyter AI also includes multiple subcommands, which may be invoked via the
`%ai` *line* magic. Jupyter AI uses subcommands to provide additional utilities

> in notebooks while keeping the same concise syntax for invoking a language model.

Reviewer: Probably OK to delete this, or to create a new small subsection about subcommands. In this new subsection, mention `%ai help`. (As of this PR, `%ai help`'s output will be small, but #136 and #170 will add to it.)

Author: I anticipate this being a major point of confusion for users, so I think it's important we note this at the outset of this section where we introduce line magics as opposed to cell magics. As you've pointed out, we're adding new subcommands very shortly, so it's still plenty relevant enough to remain where it is.

> The `%ai list` subcommand prints a list of available providers and models. Some
> providers explicitly define a list of supported models in their API. However,
> other providers, like HuggingFace Hub, lack a well-defined list of available
> models. In such cases, it's best to consult the provider's upstream
> documentation. The [HuggingFace website](https://huggingface.co/) includes a
> list of models, for example.

Reviewer: ```suggestion
The `%ai list` subcommand prints a list of available providers and models. Some
providers provide a list of supported models, and others do not.
If you don't see a list, consult the provider's documentation. The
[HuggingFace website](https://huggingface.co/) includes a
list of models, for example.
```

Author: This is a good callout for a larger documentation issue that I believe should be tackled in a separate PR. I don't like the fact that we are not exposing how the model ID is determined at all in the help documentation, and that we are exclusively forwarding them to a provider's upstream documentation. This is seldom helpful, as it doesn't offer guidance on how the local model ID is to be specified. I've created an issue to track this: https://github.com/jupyterlab/jupyter-ai/issues/198

Optionally, you can specify a provider ID as a positional argument to `%ai list`
to get all models provided by one provider. For example, `%ai list openai` will
display only models provided by the `openai` provider.

> ### Abbreviated syntax

Reviewer: Just to confirm, is this abbreviated syntax still supported? I wasn't sure about it before.

Author: Yes.
